shards/shards.go

```diff
@@ -367,51 +428,34 @@ func (ss *shardedSearcher) Search(ctx context.Context, q query.Q, opts *zoekt.Se
 		feeder <- s
 	}
 	close(feeder)
+	g := errgroup.Group{}
 	for i := 0; i < runtime.GOMAXPROCS(0); i++ {
-		go func() {
+		g.Go(func() error {
 			for s := range feeder {
-				searchOneShard(childCtx, s, q, opts, all)
-			}
-		}()
-	}
-
-	for range shards {
-		r := <-all
-		if r.err != nil {
-			return nil, r.err
-		}
-		aggregate.Files = append(aggregate.Files, r.sr.Files...)
-		aggregate.Stats.Add(r.sr.Stats)
-
-		if len(r.sr.Files) > 0 {
-			for k, v := range r.sr.RepoURLs {
-				aggregate.RepoURLs[k] = v
-			}
-			for k, v := range r.sr.LineFragments {
-				aggregate.LineFragments[k] = v
+				err := searchOneShard(childCtx, s, q, opts, stream.SenderFunc(func(sr *zoekt.SearchResult) {
+					if sr != nil {
```

Comment:
Do you need to check that SearchResult is non-nil? It seems to be we should never be passing in nil search results?

Reply:
I guess we can trust `indexedData.Search` to not return nil, nil :-).